src/containers/ServiceLocationForm.js

```diff
@@ -0,0 +1,79 @@
+/** @jsx jsx */
+import { useCallback, useState } from 'react';
+import { useTranslation } from 'react-i18next';
+import { jsx } from '@emotion/core';
+import { useHistory } from 'react-router-dom';
+
+import { Routes } from 'constants/Routes';
+import { routeWithParams } from 'lib/utils/routes';
+import { isValidZipCode } from 'lib/utils/validations';
+
+import FormBuilder from 'components/Form/FormBuilder';
+import { formFieldTypes } from 'components/Form/CreateFormFields';
+
+const validate = (val) => {
+  if (val === '') {
+    return;
+  }
+  if (!isValidZipCode(val)) {
+    return 'Please enter a valid ZIP code';
+  }
+};
+
+function ServiceLocationForm({ backend, serviceUser }) {
+  const history = useHistory();
+  const { t } = useTranslation();
+
+  const [isLoading, setIsLoading] = useState(false);
+  const [fields, setFields] = useState({
+    zipCode: serviceUser?.data?.zipCode || '',
+  });
+
+  const handleFieldChange = useCallback(
+    (field) => (value) => {
+      setFields((fields) => ({
+        ...fields,
+        [field]: value,
+      }));
+    },
+    [],
+  );
+
+  const { zipCode, ...requiredFields } = fields;
+
+  const handleSubmit = () => {
+    // service todo: save zip code field to user
+    // service todo: check valid service types
+    setIsLoading(true);
+    history.push(routeWithParams(Routes.SERVICE_TYPE));
```

Comment:
Would this be a good page to do `Routes.FACILITY`?
<img width="1199" alt="Screenshot 2020-04-17 21 55 11" src="https://user-images.githubusercontent.com/1128500/79628490-44096d00-80f6-11ea-9000-9c5864445896.png">